File: example/javaexample/src/main/java/com/github/app/javaexample/ui/IntroAdapter.java
package com.github.app.javaexample.ui;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.github.app.javaexample.R;

import java.util.List;

public class IntroAdapter extends RecyclerView.Adapter<IntroAdapter.IntroAdapterViewHolder> {

    private List<IntroEntry> entryList;
    private Context ctx;

    public IntroAdapter(List<IntroEntry> entryList, Context ctx) {
        this.entryList = entryList;
        this.ctx = ctx;
    }

    @NonNull
    @Override
    public IntroAdapterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_intro,parent,false);
        return new IntroAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull IntroAdapterViewHolder holder, int position) {
        final IntroEntry entry = entryList.get(position);
        holder.title.setText(entry.title);
        holder.description.setText(entry.description);
        holder.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(ctx, entry.activityClass);
                ctx.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return entryList.size();
    }

    public class IntroAdapterViewHolder extends RecyclerView.ViewHolder {
        TextView title;
        TextView description;
        TextView button;

        public IntroAdapterViewHolder(@NonNull View itemView) {
            super(itemView);
              title = itemView.findViewById(R.id.item_title);
              description = itemView.findViewById(R.id.item_description);
              button = itemView.findViewById(R.id.item_button);
        }
    }
}

File: example/src/main/java/com/github/appintro/example/ui/IntroAdapter.kt
package com.github.appintro.example.ui

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.github.appintro.example.R

class IntroAdapter(
        private val entries: List<IntroEntry>
) : RecyclerView.Adapter<IntroAdapter.IntroAdapterViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IntroAdapterViewHolder {
        val root = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_intro, parent, false)
        return IntroAdapterViewHolder(root)
    }

    override fun getItemCount() = entries.size

    override fun onBindViewHolder(holder: IntroAdapterViewHolder, position: Int) {
        holder.bind(entries[position])
    }

    inner class IntroAdapterViewHolder(
            private val root: View
    ) : RecyclerView.ViewHolder(root) {
        private val title: TextView = root.findViewById(R.id.item_title)
        private val description: TextView = root.findViewById(R.id.item_description)
        private val button: TextView = root.findViewById(R.id.item_button)

        fun bind(entry: IntroEntry) {
            title.setText(entry.title)
            description.setText(entry.description)
            button.setOnClickListener {
                val intent = Intent(root.context, entry.activityClass)
                root.context.startActivity(intent)
            }
        }
    }
}
